src/target/riscv/riscv.c

```diff
@@ -1189,6 +1189,33 @@ int riscv_flush_registers(struct target *target)
 	return ERROR_OK;
 }
 
+int set_debug_reason(struct target *target, enum riscv_halt_reason halt_reason)
```

Comment:
I know this is an existing code (just moved), but maybe add a note what this function does:

```suggestion
/* Convert RISC-V hart's halt reason to OpenOCD's generic debug reason */
int set_debug_reason(struct target *target, enum riscv_halt_reason halt_reason)
```